### PR TITLE
Simplify CacheLoader creation using lambdas

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by electrum** [comment]
@nezihyigitbasi Thanks, missed those by only searching for `new CacheLoader`

File: presto-local-file/pom.xml [hunk 1]
```diff
@@ -105,11 +105,5 @@
             <artifactId>testing</artifactId>
             <scope>test</scope>
         </dependency>
-
```
Comment (nezihyigitbasi) [inline presto-local-file/pom.xml]: looks like an unrelated change

File: presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java [hunk 2]
```diff
@@ -53,31 +53,29 @@
...
         this.pageFunctionCompiler = requireNonNull(pageFunctionCompiler, "pageFunctionCompiler is null");
+        this.cursorProcessors = CacheBuilder.newBuilder()
+                .recordStats()
+                .maximumSize(1000)
+                .build(CacheLoader.from(key -> compile(key.getFilter(), key.getProjections(), new CursorProcessorCompiler(metadata), CursorProcessor.class)));
         this.cacheStatsMBean = new CacheStatsMBean(cursorProcessors);
     }
 
     @Managed
+    public long getCacheSize()
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java]: move this to a separate commit ?

File: presto-local-file/pom.xml (hunk 1 above)
Comment (electrum) [reply presto-local-file/pom.xml]: It's needed because this commit removes the only usage of that dependency (`javax.annotation`).

File: presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java (hunk 2 above)
Comment (electrum) [reply presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java]: Thanks, this was a rebase conflict (another commit remove it and the rebase brought it back).
